python/src/trezorlib/cli/ethereum.py

```diff
@@ -306,13 +451,33 @@ def sign_tx(
         click.echo("Can't send tokens and custom data at the same time")
         sys.exit(1)
 
+    defs = _get_ethereum_definitions(
+        definitions_zip=definitions_zip,
+        network_def_file=network_def,
+        token_def_file=token_def,
+        download_definitions=download_definitions,
+        chain_id=chain_id,
+        token_address=to_address,
+    )
+
     address_n = tools.parse_path(address)
-    from_address = ethereum.get_address(client, address_n)
+    from_address = ethereum.get_address(
+        client, address_n, encoded_network=defs.encoded_network
+    )
 
     if token:
         data = _erc20_contract(token, to_address, amount)
         to_address = token
         amount = 0
+        # to_address has changed, reload definitions
```

Comment:
Not completely sure here, but if we moved `if token` block (just those 4 lines without defs recalculation) above the first `defs = _get_ethereum_definitions` calculation, then it would not be necessary to reload the definitions again.

Only result from `defs` we use in between is `defs.encoded_network` and I guess this cannot change with the change of `to_address`.

We would therefore save the need of reloading.

Reply:
affected code was changed